crates/compiler/builtins/roc/Num.roc

```diff
@@ -867,6 +868,41 @@ max = \a, b ->
     else
         b
 
+## Restricts a number to be between two other numbers.
+##
+## ```
+## restrictToInterval 5 { startAt: 3, endAt: 8 } == 5
+## restrictToInterval 1 { startAt: 3, endAt: 8 } == 3
+## restrictToInterval 10 { startAt: 3, endAt: 8 } == 8
+
+## restrictToInterval 5 { startAt: 8, endAt: 3 } == 5
+## restrictToInterval 1 { startAt: 8, endAt: 3 } == 3
+## restrictToInterval 10 { startAt: 8, endAt: 3 } == 8
+## ```
+## You may know this function as `clamp` in other languages.
+restrictToInterval : Num a, { startAt : Num a, endAt : Num a } -> Num a
+restrictToInterval = \x, { startAt, endAt } ->
+    if endAt > startAt then
+        restrictToInterval x { startAt: endAt, endAt: startAt }
```

Comment:
This doesn't seem right to me. Did you mean `endAt < startAt`?

Separately, I wonder if that's the desired behavior. Maybe we should just clamp to the `startAt` value in this case?